src/schema/resolvers.js

```diff
@@ -7,19 +7,140 @@ const resolvers = {
   Date: DateScalar,
 
   Query: {
-    getUserById: async (_, args) => {
+    getUserById: async (_, { userId }) => {
       const user = await prisma.user.findUnique({
-        where: { userId: args.id },
+        where: { userId: userId },
+
+        // Include the relationship data
+        include: {
+          skills: true,
+          jobApplications: true,
+          jobPostings: true,
+          contacts: true,
+          resume: true,
+        },
       })
 
       if (!user) throw new Error('That user does not exist')
 
       return user
     },
 
-    searchUsers: async (_, args) => {
+    searchUsers: async (_, { name }) => {
       return await prisma.user.findMany({
-        where: { name: args.name },
+        where: {
+          name: {
+            contains: name,
+          },
+        },
+
+        include: {
+          skills: true,
+          jobApplications: true,
+          jobPostings: true,
+          contacts: true,
+        },
+      })
+    },
+
+    getJobApplicationById: async (_, { jobAppId }) => {
+      const jobApplication = await prisma.jobApplication.findUnique({
+        where: { jobAppId: jobAppId },
+      })
+
+      if (!jobApplication)
+        throw new Error('That job application does not exist')
```

Comment:
This applies to all `getThingById`

I don't think we should throw an error if an entity is not found. An error gives the impression that something bad/unexpected happened. Not being able to find something by the given id is something I would consider normal behavior. The response for these queries is also nullable in our schema so we should be fine returning null if that's what `findUnique` returns

Reply:
I removed the error handling from getThingsById. I added some to the mutations.